src/App.tsx

```diff
@@ -1,14 +1,67 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { FilterBy, Todo } from './types/Todo';
+import { User } from './types/User';
+import { filterGoods } from './todo/todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.ALL);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then((todosData: Todo[]) => {
+        setTodos(todosData);
+      })
+      .catch(() => setErrorMessage('Try again later'))
+      .finally(() => setLoading(false));
+  }, []);
+
+  useEffect(() => {
+    if (!selectedTodoId) {
+      setSelectedTodo(null);
+
+      return;
+    }
+
+    const openedTodo = todos.find(todo => todo.id === selectedTodoId);
+
+    if (openedTodo?.userId) {
+      setLoading(true);
```

Comment:
it's not correct to have one `loading` for users and todos. It's different entities